ui/src/components/ConfigurationForm/ConfigurationForm.component.tsx

```diff
@@ -155,7 +154,7 @@ export const ConfigurationForm: React.FC<ConfigurationFormProps> = ({ isEmbedded
             name="implementation"
             label="Server implementation"
             disabled={!isEffectTypeSelected}
-            options={isEffectTypeSelected ? getEffectImplementationOptions(effectType) : []}
+            options={isEffectTypeSelected ? getEffectImplementationOptions(effectType, scalaVer) : []}
```

Comment:
Ok, so in my opinion here is how server implementation select should behave. Currently we disable this field if user did not select any effect type, with the recent change of adding a scala version, effect implementation has become also dependent on that new value.

so const `isEffectTypeSelected` should be renamed to something like:
```typescript
const isEffectImplementationSelectable = Boolean(effectType) && Boolean(scalaVer);

...

<FormSelect
  className={classes.formEffectsRow}
  name="implementation"
  label="Server implementation"
  disabled={!isEffectImplementationSelectable}
  options={isEffectImplementationSelectable ? getEffectImplementationOptions(effectType, scalaVer) : []}
/>
```
we should disable the select until user selects both effect type and scala version, we want to mitigate scenarios like below, for `Future` effect type user cannot select `Akka` because he hasn't selected a scala version. For someone who is here first time this is a discoverability problem.
![2022-08-01-151634_1031x654_scrot](https://user-images.githubusercontent.com/24817919/182156642-7bd4f8e9-6e7a-45f0-96b7-2fdccb17ffe1.png)

Reply:
Additionally we have to update the `useEffect` hook in order to reset server implementation field upon scala version change:
```typescript
useEffect(() => {
  ...
  // NOTE: reset effect implementation field value upon scala version change
  if (
    scalaVer &&
    scalaVer === ScalaVersion.Scala3 &&
    effectImplementation &&
    forbiddenScala3EffectImplementations.includes(effectImplementation)
  ) {
    form.resetField('implementation');
  }
  ...
}, [effectType, scalaVer, effectImplementation, jsonImplementation, form]);
```

Reply:
Thx! I haven't noticed reset functionality!  Done.